src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -411,12 +411,25 @@ open_locale arithmetic_function
 
 /-- `cyclotomic n R` can be expressed as a product in a fraction field of `polynomial R`
   using Möbius inversion. -/
-lemma cyclotomic_eq_prod_X_pow_sub_one_pow_moebius {n : ℕ} (hpos : 0 < n)
-  (R : Type*) [comm_ring R] [nontrivial R]
+lemma cyclotomic_eq_prod_X_pow_sub_one_pow_moebius {n : ℕ} (R : Type*) [comm_ring R]
   {K : Type*} [field K] [algebra (polynomial R) K] [is_fraction_ring (polynomial R) K] :
```

Comment:
I think that these assumptions force `R` to be an integral domain.
@Vierkantor should this now use `ratfunc` instead?

Reply:
Indeed, if `R` has zero divisors then `(algebra_map R K).map_dvd` shows `K` can't be a field: it is either trivial or has zero divisors.

Using `ratfunc` would result in a neater statement, the downside being that the current version is generic over all implementations of `R(x)`. Let me see how much we'll need to change the proof...